Update workflow_utils build modules and remove ncio module hack

KateFriedman-NOAA opened the pull request:
This PR removes the ncio module hack in the workflow_utils build and updates the workflow_utils build modules on all supported platforms to add the ncio module available via hpc-stack.

Close #407

KateFriedman-NOAA committed:
Add ncio/1.0.0, landsfcutil/2.4.1, and wgrib2/2.0.8 modules to workflow_utils for Jet

Refs: #407

KateFriedman-NOAA committed:
Remove ncio library hack from workflow_utils build

Refs: #407

KateFriedman-NOAA committed:
Add ncio/1.0.0, landsfcutil/2.4.1, and wgrib2/2.0.8 modules to workflow_utils for WCOSS-Dell, Hera, Orion

Refs; #407

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/issue407

KateFriedman-NOAA committed:
Adjust workflow_utils modulefiles for all platforms

- remove unneeded landsfcutil/2.4.1 and wgrib2/2.0.8 modules
- move hdf5/1.10.6 and netcdf/4.7.4 module loads before ncio module load;
  ncio module prereq in hpc-stack is netcdf module
- workflow_utils build retested successfully on WCOSS-Dell, Hera, Jet, and Orion

Refs: #407

KateFriedman-NOAA commented:
Completing 2.5 cycle C192C96L127 test on WCOSS-Dell with final PR branch before submitting this after approval.

aerorahul commented on modulefiles/workflow_utils.hera:
```diff
@@ -20,10 +20,9 @@ module load sp/2.3.3
 module load nemsio/2.5.2
 module load nemsiogfs/2.5.3
-#module load ncio/1.0.0
+module load hdf5/1.10.6
+module load netcdf/4.7.4
```
please move this right after `png`.
One of the `w3` libraries used to have netcdf dependency (or may be still does).

KateFriedman-NOAA replied:
Done

aerorahul commented on modulefiles/workflow_utils.jet:
```diff
@@ -24,9 +24,9 @@ module load sp/2.3.3
 module load ip/3.3.3
 module load nemsio/2.5.2
 module load nemsiogfs/2.5.3
+module load hdf5/1.10.6
+module load netcdf/4.7.4
```
same comment as above.

KateFriedman-NOAA replied:
Done

aerorahul commented on modulefiles/workflow_utils.orion:
```diff
@@ -20,10 +20,9 @@ module load sp/2.3.3
 module load nemsio/2.5.2
 module load nemsiogfs/2.5.3
-#module load ncio/1.0.0
+module load hdf5/1.10.6
+module load netcdf/4.7.4
```
same here.

KateFriedman-NOAA replied:
Done

aerorahul commented on modulefiles/workflow_utils.wcoss_dell_p3:
```diff
@@ -20,10 +20,9 @@ module load sp/2.3.3
 module load nemsio/2.5.2
 module load nemsiogfs/2.5.3
-#module load ncio/1.0.0
+module load hdf5/1.10.6
+module load netcdf/4.7.4
```
and here.

KateFriedman-NOAA replied:
Done

aerorahul reviewed (changes requested):
Looks good.
One small request for change.

KateFriedman-NOAA committed:
Move hdf5 and netcdf module load order up in workflow_utils build modulefiles

- move the hdf5/1.10.6 and netcdf/4.7.4 module loads earlier in workflow_utils modulefiles
- loads these modules as prereqs for other modules loaded later in modulefile

Refs: #407

KateFriedman-NOAA commented:
Moved hdf5 and netcdf module loads earlier in workflow_utils modulefiles since they are prereqs for some other modules loaded later in modulefiles. Retested workflow_utils build on WCOSS-Dell, Hera, Jet, and Orion - successful.

aerorahul reviewed (approved):
looks good.
thanks for removing hacks.
